tcms/rpc/api/testexecution.py

```diff
@@ -183,12 +183,25 @@ def update(execution_id, values, **kwargs):
 
     # if this call updated TE.status then adjust timestamps
     if values.get("status"):
+        now = timezone.now()
         if test_execution.status.weight != 0:
-            test_execution.close_date = timezone.now()
+            test_execution.close_date = now
         else:
             test_execution.close_date = None
         test_execution.save()
 
+        all_runs = TestExecution.objects.filter(run=test_execution.run)
+        if test_execution.status.weight > 0 or test_execution.status.weight < 0:
+            has_in_progress = all_runs.filter(status__weight=0).exists()
+            if not has_in_progress:
+                run = TestRun.objects.get(pk=test_execution.run.pk)
+                run.stop_date = now
+                run.save()
+        else:
```

Comment:
this should be `test_execution.status.weight == 0 and not test_execution.run.stop_date`

Reply:
why? if `not test_execution.run.stop_date` is true, then `test_execution.run.stop_date = None` one line below is no-op because it does not make any changes?

Reply:
> no-op because it does not make any changes

This is simply not true. 

From what I can tell `.save()` will update the DB with the same values. This will cause all sorts of side effects along the line, most notably history entries without any diff and bunch of signals sent.  Experiment with:
```
>>> tr.stop_date = tr.stop_date
>>> tr.save()
```

Reply:
ok, but when I tested this locally it did not achieved what we wanted it to (or I don't understand what we want to achieve)